test_unstructured_inference/test_utils.py

```diff
@@ -0,0 +1,44 @@
+import pytest
+from unstructured_inference.utils import LazyDict, LazyEvaluateInfo
+
+
+def test_dict_same():
+    d = {"a": 1, "b": 2, "c": 3}
+    ld = LazyDict(**d)
+    assert all(kd == kld for kd, kld in zip(d, ld))
+    assert all(d[k] == ld[k] for k in d)
+    assert len(ld) == len(d)
+
+
+def test_lazy_evaluate():
+    called = 0
+
+    def func(x):
+        nonlocal called
```

Comment:
Interesting. I knew about `global` before but didn't realize there was a `nonlocal` too

Reply:
Yeah I had trouble remember how to get the scope one level up and had to look it up.